ClickHouseConnector.pq

```diff
@@ -475,6 +482,48 @@ ClickHouse.Icons = [
 ];
 
 
+ConvertOptionsStringToRecord = (optionsString as text, ValidOptionsMap as table) as record =>
+        let
+            // Split the options string into rows
+            optionRows = Text.Split(optionsString, ";"),
+
+            // Split each row into key-value pairs
+            keyValuePairs = List.Transform(optionRows, each Text.Split(_, "=")),
+
+            flattenedList = List.Combine(keyValuePairs),
+
+            extractedOptionsFromMap = List.Select(ValidOptionsMap[Name], each List.Contains(keyValuePairs, {_, _})),
+            // Identify keys from optionsString that don't exist in ValidOptionsMap
+        invalidKeys = List.Difference(List.Transform(keyValuePairs, each Text.Clean(Text.Trim(_{0}))), ValidOptionsMap[Name]),
+        extractedOptionsRecord = Record.FromList(List.Alternate(flattenedList, 1, 1), List.Alternate(flattenedList, 1, 1, 1)),
+        
+
+        optionsToTransform = Table.SelectRows(ValidOptionsMap, each (not Type.Is([Type],type nullable text) and not Type.Is([Type],type text))),
```

Comment:
Is it possible to test the logic somehow?

Reply:
Sure, I tested the following points:
* I ensured that the connection string passed to the ODBC connector was accurate (the right options were included).
* I tested that I get a connection timeout when querying a local HTTP microservice